src/library/autodj/autodjprocessor.cpp

```diff
@@ -532,30 +528,37 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
         }
     }
 
-    if (m_eState == ADJ_IDLE && thisDeck.isRepeat()) {
-        // repeat pauses auto DJ
-        return;
+    if (m_eState == ADJ_IDLE) {
+        if (!thisDeckPlaying) {
+            // this is a cueing seek, recalculate the transition, from the
+            // new position
+            calculateTransition(&otherDeck, &thisDeck, false, false);
```

Comment:
This seems to work from my brief testing, but the flow of the code is difficult to follow. If I understand correctly, when seeking the "to deck", calculateTransition gets called, the deck is seeked, then because of the seek playerPositionChanged gets called and calculateTransition gets called again at this line. This is a roundabout way to handle this. Is there a way to make the code more straightforward? Maybe do the seeking in calculateTransition and remove this line in playerPositionChanged?

Reply:
Yes, you are right. In this case the calculateTransition is called twice. 
We need to keep this call here to adjust the times after manual seeks from the user. Unfortunately this slot is called asynchrony. To avoid this we may add a global flag to skip this seek but that feels hacky. This solution has the "advantage" that the transition is refreshed after the seek. 

I have just noticed a bug that the seek after track load is skipped. I need to have a look.

Reply:
If we decide to keep the code how it is (or close to it), at the least we need comments explaining how this works.

Reply:
It is because the seek from Auto DJ is a direct connection while the initial seek from the cue point is delivered as queued connection. The later is indicated on the waveforms.

Reply:
> If we decide to keep the code how it is (or close to it), at the least we need comments explaining how this works.

Yes, I will try.